userspace/libsinsp/runc.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright (C) 2013-2019 Draios Inc dba Sysdig.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "runc.h"
+
+#include <cstring>
+
+#include "sinsp.h"
+#include "sinsp_int.h"
+
+namespace {
+
+const size_t CONTAINER_ID_LENGTH = 64;
+const size_t REPORTED_CONTAINER_ID_LENGTH = 12;
+const char* CONTAINER_ID_VALID_CHARACTERS = "0123456789abcdefABCDEF";
+
+static_assert(REPORTED_CONTAINER_ID_LENGTH <= CONTAINER_ID_LENGTH, "Reported container ID length cannot be longer than actual length");
+
+bool detect_one_container_id(const std::string& cgroup, const std::string& prefix, const std::string& suffix, std::string& container_id)
+{
+	size_t pos = cgroup.rfind(prefix);
```

Comment:
we're screwed if the container ID contains the prefix...unless the assumption is that the prefix will always contain a special character which can appear nowhere later in the string. If that is the case we should make that explicit

Reply:
The prefix will never appear in the id or suffix _for now_ (there's a kube-proxy patch on the horizon which will change that but we don't know everything yet). I'd rather hold off functional changes until we investigate the kube-proxy issue (it'll either be a smallish change, including your proposed fix, or a complete redesign of the matching algorithm).

Reply:
so long as there's a bug, that's fine. if we could leave a note here in case that change never makes it, that would be great